kv/kvserver: TestRangeInfo failed

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1814032&tab=buildLog) on [master@aa18c27ce78016377abe9fbcb3ca386b4eca4887](https://github.com/cockroachdb/cockroach/commits/aa18c27ce78016377abe9fbcb3ca386b4eca4887):

```
=== RUN   TestRangeInfo
I200317 19:17:14.005511 238174 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38745" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200317 19:17:14.100814 238174 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200317 19:17:14.101907 238174 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:41645" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 19:17:14.119179 238304 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:38745
I200317 19:17:14.291989 238174 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200317 19:17:14.309346 238174 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=bc68ceab] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200317 19:17:14.359420 238174 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot d75c6dbe at applied index 19
I200317 19:17:14.362938 238174 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200317 19:17:14.383641 238602 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=d75c6dbe index=19]
I200317 19:17:14.397045 238602 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=6ms ingestion=4@2ms id=d75c6dbe index=19]
I200317 19:17:14.444991 238174 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200317 19:17:14.462201 238174 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=89045771] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200317 19:17:14.574946 238174 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200317 19:17:15.161641 238855 util/stop/stopper.go:539  quiescing
I200317 19:17:15.163795 238856 util/stop/stopper.go:539  quiescing
W200317 19:17:15.174591 238305 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200317 19:17:15.174985 238305 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
I200317 19:17:15.184384 238515 kv/kvserver/queue.go:578  [s2,r2/2:{a-/Max}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200317 19:17:15.187762 238305 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200317 19:17:15.188128 238305 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
I200317 19:17:15.219343 238854 util/stop/stopper.go:539  quiescing
I200317 19:17:15.231565 238854 util/stop/stopper.go:539  quiescing
I200317 19:17:15.233590 238854 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.33s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584472635.328331311,0 pro=1584472634.428465004,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584472635.328331311,0 pro=1584472634.428465004,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584472635.778975231,0 pro=1584472634.879065894,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584472635.328331311,0 pro=1584472634.428465004,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584472635.328331311,0 pro=1584472634.428465004,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584472635.328331311,0 pro=1584472634.428465004,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584472635.328331311,0 pro=1584472634.428465004,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584472635.778975231,0 pro=1584472634.879065894,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1814544&tab=buildLog) on [master@b5f030223fbcf22e806c48a3c46e74a73a54f50f](https://github.com/cockroachdb/cockroach/commits/b5f030223fbcf22e806c48a3c46e74a73a54f50f):

```
=== RUN   TestRangeInfo
I200318 05:48:12.674749 239486 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:35497" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200318 05:48:12.843816 239486 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200318 05:48:12.845045 239486 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:40109" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 05:48:12.856119 239666 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:35497
I200318 05:48:12.911705 239486 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200318 05:48:12.929510 239486 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=3a3c86c3] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200318 05:48:12.970021 239486 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 9a9a60e9 at applied index 18
I200318 05:48:12.972671 239486 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200318 05:48:12.995273 239866 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200318 05:48:12.997541 239818 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=9a9a60e9 index=18]
I200318 05:48:13.009133 239818 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=10ms ingestion=4@6ms id=9a9a60e9 index=18]
I200318 05:48:13.029748 239486 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200318 05:48:13.058563 239486 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=54a5686c] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200318 05:48:13.101272 239486 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 05:48:13.537101 240014 util/stop/stopper.go:539  quiescing
I200318 05:48:13.537688 240013 util/stop/stopper.go:539  quiescing
I200318 05:48:13.587581 240117 util/stop/stopper.go:539  quiescing
W200318 05:48:13.596842 239823 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200318 05:48:13.597086 239843 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200318 05:48:13.599985 239877 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200318 05:48:13.602097 240022 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200318 05:48:13.606867 240117 util/stop/stopper.go:539  quiescing
I200318 05:48:13.611492 240117 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.05s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584510493.573132365,0 pro=1584510492.673542358,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584510493.573132365,0 pro=1584510492.673542358,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584510494.065512632,0 pro=1584510493.166350948,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584510493.573132365,0 pro=1584510492.673542358,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584510493.573132365,0 pro=1584510492.673542358,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584510493.573132365,0 pro=1584510492.673542358,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584510493.573132365,0 pro=1584510492.673542358,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584510494.065512632,0 pro=1584510493.166350948,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1815929&tab=buildLog) on [master@b5f030223fbcf22e806c48a3c46e74a73a54f50f](https://github.com/cockroachdb/cockroach/commits/b5f030223fbcf22e806c48a3c46e74a73a54f50f):

```
=== RUN   TestRangeInfo
I200318 17:14:58.633233 242278 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:45995" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200318 17:14:58.748299 242278 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200318 17:14:58.749322 242278 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:45413" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 17:14:58.754626 242490 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:45995
I200318 17:14:58.876200 242278 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200318 17:14:58.901943 242278 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=c1522bfd] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200318 17:14:58.948227 242278 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot c67c39a3 at applied index 18
I200318 17:14:58.956376 242278 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.04s
I200318 17:14:58.966482 242629 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=c67c39a3 index=18]
I200318 17:14:58.972552 242629 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=5ms ingestion=4@2ms id=c67c39a3 index=18]
I200318 17:14:59.021044 242278 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200318 17:14:59.068303 242278 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=fc8b1c79] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200318 17:14:59.328952 242278 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 17:14:59.847832 242813 util/stop/stopper.go:539  quiescing
I200318 17:14:59.851371 242812 util/stop/stopper.go:539  quiescing
I200318 17:14:59.857524 242290 kv/kvserver/queue.go:578  [s2,r2/2:{a-/Max}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200318 17:14:59.861551 242510 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 17:14:59.909336 242510 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200318 17:14:59.932998 242811 util/stop/stopper.go:539  quiescing
W200318 17:14:59.939568 242434 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200318 17:14:59.941018 242707 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200318 17:14:59.943607 242703 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200318 17:14:59.946025 242811 util/stop/stopper.go:539  quiescing
I200318 17:14:59.947868 242811 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.41s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584551699.969164475,0 pro=1584551699.069620785,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584551699.969164475,0 pro=1584551699.069620785,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584551700.445900622,0 pro=1584551699.547199212,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584551699.969164475,0 pro=1584551699.069620785,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584551699.969164475,0 pro=1584551699.069620785,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584551699.969164475,0 pro=1584551699.069620785,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584551699.969164475,0 pro=1584551699.069620785,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584551700.445900622,0 pro=1584551699.547199212,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816324&tab=buildLog) on [master@37f2a6964760b5d72bef655342628e31601b8978](https://github.com/cockroachdb/cockroach/commits/37f2a6964760b5d72bef655342628e31601b8978):

```
=== RUN   TestRangeInfo
I200318 20:35:00.158293 238047 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:43645" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200318 20:35:00.266276 238047 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200318 20:35:00.267421 238047 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:40829" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 20:35:00.272257 238249 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:43645
I200318 20:35:00.321578 238047 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200318 20:35:00.338194 238047 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=ebdaf72f] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200318 20:35:00.384689 238047 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot ca7f50d3 at applied index 18
I200318 20:35:00.388636 238047 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200318 20:35:00.422212 238445 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=ca7f50d3 index=18]
I200318 20:35:00.423057 238418 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200318 20:35:00.432213 238445 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=9ms ingestion=4@7ms id=ca7f50d3 index=18]
I200318 20:35:00.446277 238047 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200318 20:35:00.461271 238047 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=e7a7d6fe] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200318 20:35:00.522279 238047 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 20:35:01.058223 238694 util/stop/stopper.go:539  quiescing
I200318 20:35:01.058355 238695 util/stop/stopper.go:539  quiescing
I200318 20:35:01.117969 238693 util/stop/stopper.go:539  quiescing
W200318 20:35:01.121816 238471 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200318 20:35:01.124462 238319 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200318 20:35:01.126538 238455 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200318 20:35:01.128406 238583 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200318 20:35:01.135910 238693 util/stop/stopper.go:539  quiescing
I200318 20:35:01.137832 238693 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.06s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584563701.043502458,0 pro=1584563700.143601719,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584563701.043502458,0 pro=1584563700.143601719,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584563701.500776240,0 pro=1584563700.600875069,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584563701.043502458,0 pro=1584563700.143601719,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584563701.043502458,0 pro=1584563700.143601719,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584563701.043502458,0 pro=1584563700.143601719,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584563701.043502458,0 pro=1584563700.143601719,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584563701.500776240,0 pro=1584563700.600875069,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816512&tab=buildLog) on [master@47e5748e864f37ae9eca6abbfbb04a7957662b50](https://github.com/cockroachdb/cockroach/commits/47e5748e864f37ae9eca6abbfbb04a7957662b50):

```
=== RUN   TestRangeInfo
I200318 22:25:33.925871 241237 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:45677" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200318 22:25:34.033037 241237 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200318 22:25:34.035331 241237 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:35331" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 22:25:34.039467 241462 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:45677
I200318 22:25:34.177411 241237 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200318 22:25:34.204280 241237 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=269eb4cd] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200318 22:25:34.245315 241237 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 87a6db96 at applied index 18
I200318 22:25:34.253354 241237 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200318 22:25:34.312004 241526 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=87a6db96 index=18]
I200318 22:25:34.318943 241552 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200318 22:25:34.333086 241561 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200318 22:25:34.344574 241526 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=32ms ingestion=4@15ms id=87a6db96 index=18]
I200318 22:25:34.359889 241237 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200318 22:25:34.409965 241237 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=91c7ec21] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200318 22:25:34.545857 241237 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 22:25:35.188961 241684 util/stop/stopper.go:539  quiescing
I200318 22:25:35.189711 241683 util/stop/stopper.go:539  quiescing
I200318 22:25:35.190129 241417 kv/kvserver/queue.go:578  [s2,r2/2:{a-/Max}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200318 22:25:35.238874 241682 util/stop/stopper.go:539  quiescing
W200318 22:25:35.242460 241470 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 22:25:35.249660 241671 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 22:25:35.254221 241557 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
I200318 22:25:35.257051 241682 util/stop/stopper.go:539  quiescing
I200318 22:25:35.259036 241682 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.43s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584570335.265945424,0 pro=1584570334.366062777,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584570335.265945424,0 pro=1584570334.366062777,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584570335.716053464,0 pro=1584570334.816165495,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584570335.265945424,0 pro=1584570334.366062777,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584570335.265945424,0 pro=1584570334.366062777,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584570335.265945424,0 pro=1584570334.366062777,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584570335.265945424,0 pro=1584570334.366062777,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584570335.716053464,0 pro=1584570334.816165495,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816554&tab=buildLog) on [master@37f2a6964760b5d72bef655342628e31601b8978](https://github.com/cockroachdb/cockroach/commits/37f2a6964760b5d72bef655342628e31601b8978):

```
=== RUN   TestRangeInfo
I200318 22:40:39.706315 237825 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:37621" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200318 22:40:39.820008 237825 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200318 22:40:39.821075 237825 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43625" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 22:40:39.833729 238075 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:37621
I200318 22:40:39.903162 237825 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200318 22:40:39.916020 237825 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=5f5375e7] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200318 22:40:39.949600 237825 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 7e5e24db at applied index 18
I200318 22:40:39.952305 237825 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200318 22:40:39.960361 238033 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=7e5e24db index=18]
I200318 22:40:39.966361 238033 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=6ms ingestion=4@2ms id=7e5e24db index=18]
I200318 22:40:39.976331 237825 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200318 22:40:40.003508 237825 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=27448558] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200318 22:40:40.074481 237825 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 22:40:40.532889 238457 util/stop/stopper.go:539  quiescing
I200318 22:40:40.533948 238458 util/stop/stopper.go:539  quiescing
I200318 22:40:40.538661 238446 kv/kvserver/queue.go:1189  [replicate] purgatory is now empty
I200318 22:40:40.595646 238456 util/stop/stopper.go:539  quiescing
W200318 22:40:40.610151 238315 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 22:40:40.613236 238212 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 22:40:40.613400 238281 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200318 22:40:40.614899 238091 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200318 22:40:40.616349 238456 util/stop/stopper.go:539  quiescing
I200318 22:40:40.618105 238456 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (0.98s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584571240.591440119,0 pro=1584571239.691558592,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584571240.591440119,0 pro=1584571239.691558592,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584571241.042327513,0 pro=1584571240.142431947,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584571240.591440119,0 pro=1584571239.691558592,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584571240.591440119,0 pro=1584571239.691558592,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584571240.591440119,0 pro=1584571239.691558592,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584571240.591440119,0 pro=1584571239.691558592,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584571241.042327513,0 pro=1584571240.142431947,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816742&tab=buildLog) on [master@a16d8aa11a5248a7324333bd60fbfd749ca6bb2d](https://github.com/cockroachdb/cockroach/commits/a16d8aa11a5248a7324333bd60fbfd749ca6bb2d):

```
=== RUN   TestRangeInfo
I200319 00:25:08.241401 241255 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:43547" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200319 00:25:08.395979 241255 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200319 00:25:08.401681 241255 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:38181" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200319 00:25:08.405972 241187 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:43547
I200319 00:25:08.508457 241255 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200319 00:25:08.523501 241255 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=ff716c1a] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200319 00:25:08.550081 241255 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot f5681fbf at applied index 18
I200319 00:25:08.552832 241255 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200319 00:25:08.564472 241592 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=f5681fbf index=18]
I200319 00:25:08.576839 241592 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=12ms ingestion=4@7ms id=f5681fbf index=18]
I200319 00:25:08.638293 241255 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200319 00:25:08.660522 241255 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=61a6155b] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200319 00:25:08.821970 241255 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200319 00:25:09.573831 241929 util/stop/stopper.go:539  quiescing
I200319 00:25:09.574767 241928 util/stop/stopper.go:539  quiescing
I200319 00:25:09.622814 241927 util/stop/stopper.go:539  quiescing
W200319 00:25:09.633666 241621 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200319 00:25:09.634390 241711 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200319 00:25:09.634805 241716 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
I200319 00:25:09.645543 241927 util/stop/stopper.go:539  quiescing
I200319 00:25:09.647921 241927 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.51s)
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584577509.566701979,0 pro=1584577508.666780456,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584577509.566701979,0 pro=1584577508.666780456,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584577509.566701979,0 pro=1584577508.666780456,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584577510.017885881,0 pro=1584577509.117970672,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816815&tab=buildLog) on [master@1ae9b8d2cd23fe97fb2a7e81c73639eb308ee502](https://github.com/cockroachdb/cockroach/commits/1ae9b8d2cd23fe97fb2a7e81c73639eb308ee502):

```
=== RUN   TestRangeInfo
I200319 01:48:58.697761 237228 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42613" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200319 01:48:58.775423 237228 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200319 01:48:58.776408 237228 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:40995" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200319 01:48:58.786071 237608 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:42613
I200319 01:48:58.860789 237228 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200319 01:48:58.871776 237228 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=13483bc4] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200319 01:48:58.891049 237228 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 263f496d at applied index 18
I200319 01:48:58.909909 237228 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200319 01:48:58.925128 237704 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=263f496d index=18]
I200319 01:48:58.940913 237704 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=15ms ingestion=4@12ms id=263f496d index=18]
I200319 01:48:58.952663 237228 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200319 01:48:58.996444 237228 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=f1808904] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200319 01:48:59.093958 237228 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200319 01:48:59.440072 237904 util/stop/stopper.go:539  quiescing
I200319 01:48:59.440797 237903 util/stop/stopper.go:539  quiescing
I200319 01:48:59.492768 237902 util/stop/stopper.go:539  quiescing
W200319 01:48:59.497177 237678 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200319 01:48:59.498519 237845 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200319 01:48:59.500455 237676 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200319 01:48:59.501632 237902 util/stop/stopper.go:539  quiescing
I200319 01:48:59.503061 237902 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (0.89s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584582539.586447539,0 pro=1584582538.686728790,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584582539.586447539,0 pro=1584582538.686728790,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584582540.082907164,0 pro=1584582539.185401650,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584582539.586447539,0 pro=1584582538.686728790,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584582539.586447539,0 pro=1584582538.686728790,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584582539.586447539,0 pro=1584582538.686728790,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584582539.586447539,0 pro=1584582538.686728790,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584582540.082907164,0 pro=1584582539.185401650,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816955&tab=buildLog) on [master@d571624ae4f833c49c717728a74cc7be78a791f0](https://github.com/cockroachdb/cockroach/commits/d571624ae4f833c49c717728a74cc7be78a791f0):

```
=== RUN   TestRangeInfo
I200319 06:07:16.400992 237898 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34231" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200319 06:07:16.494135 237898 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200319 06:07:16.495339 237898 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:40181" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200319 06:07:16.499030 238041 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:34231
I200319 06:07:16.568150 237898 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200319 06:07:16.579898 237898 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=6d032cd5] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200319 06:07:16.610736 237898 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 9f8b17fa at applied index 18
I200319 06:07:16.627377 237898 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200319 06:07:16.635884 238140 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=9f8b17fa index=18]
I200319 06:07:16.642544 238140 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=6ms ingestion=4@3ms id=9f8b17fa index=18]
I200319 06:07:16.656667 237898 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200319 06:07:16.678146 237898 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=6926b935] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200319 06:07:16.719380 237898 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200319 06:07:17.154762 238517 util/stop/stopper.go:539  quiescing
I200319 06:07:17.156438 238518 util/stop/stopper.go:539  quiescing
W200319 06:07:17.160671 238124 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200319 06:07:17.219205 238516 util/stop/stopper.go:539  quiescing
W200319 06:07:17.226799 238173 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200319 06:07:17.230514 238312 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200319 06:07:17.234424 238516 util/stop/stopper.go:539  quiescing
I200319 06:07:17.236402 238516 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (0.90s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584598037.289708128,0 pro=1584598036.389846810,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584598037.289708128,0 pro=1584598036.389846810,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584598037.741187921,0 pro=1584598036.841295832,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584598037.289708128,0 pro=1584598036.389846810,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584598037.289708128,0 pro=1584598036.389846810,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584598037.289708128,0 pro=1584598036.389846810,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584598037.289708128,0 pro=1584598036.389846810,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584598037.741187921,0 pro=1584598036.841295832,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1818434&tab=buildLog) on [master@d571624ae4f833c49c717728a74cc7be78a791f0](https://github.com/cockroachdb/cockroach/commits/d571624ae4f833c49c717728a74cc7be78a791f0):

```
=== RUN   TestRangeInfo
I200319 08:25:42.894809 242866 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38301" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200319 08:25:42.972864 242866 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200319 08:25:42.974191 242866 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:36953" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200319 08:25:42.982896 243317 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:38301
I200319 08:25:43.057080 242866 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200319 08:25:43.067200 242866 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=53288ab5] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200319 08:25:43.125635 242866 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 95a0cf1c at applied index 18
I200319 08:25:43.128937 242866 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200319 08:25:43.137963 243387 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200319 08:25:43.142879 243357 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=95a0cf1c index=18]
I200319 08:25:43.156447 243357 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=11ms ingestion=4@5ms id=95a0cf1c index=18]
I200319 08:25:43.176956 242866 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200319 08:25:43.195163 242866 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=0f8bd6cb] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200319 08:25:43.261397 242866 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200319 08:25:43.753301 243609 util/stop/stopper.go:539  quiescing
I200319 08:25:43.760437 243610 util/stop/stopper.go:539  quiescing
I200319 08:25:43.806226 243608 util/stop/stopper.go:539  quiescing
W200319 08:25:43.815228 243214 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200319 08:25:43.815581 243371 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200319 08:25:43.818570 243509 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
I200319 08:25:43.821513 243608 util/stop/stopper.go:539  quiescing
I200319 08:25:43.823381 243608 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.00s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584606343.777948616,0 pro=1584606342.878097734,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584606343.777948616,0 pro=1584606342.878097734,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584606344.237626121,0 pro=1584606343.339101868,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584606343.777948616,0 pro=1584606342.878097734,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584606343.777948616,0 pro=1584606342.878097734,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584606343.777948616,0 pro=1584606342.878097734,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584606343.777948616,0 pro=1584606342.878097734,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584606344.237626121,0 pro=1584606343.339101868,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1818931&tab=buildLog) on [master@37640ca03ccaaf0192b77198722a75d454f81885](https://github.com/cockroachdb/cockroach/commits/37640ca03ccaaf0192b77198722a75d454f81885):

```
=== RUN   TestRangeInfo
I200319 18:12:36.850634 240297 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:44639" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200319 18:12:37.007578 240297 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200319 18:12:37.008864 240297 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:33553" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200319 18:12:37.016649 240414 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44639
I200319 18:12:37.120367 240297 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200319 18:12:37.156760 240297 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=f4656866] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200319 18:12:37.213435 240297 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 638dec6b at applied index 19
I200319 18:12:37.218895 240297 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200319 18:12:37.230204 240462 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=638dec6b index=19]
I200319 18:12:37.243408 240462 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=12ms ingestion=4@6ms id=638dec6b index=19]
I200319 18:12:37.273264 240297 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200319 18:12:37.293933 240297 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=41fc3d89] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200319 18:12:37.495852 240297 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200319 18:12:38.126266 240855 util/stop/stopper.go:539  quiescing
I200319 18:12:38.126953 240854 util/stop/stopper.go:539  quiescing
W200319 18:12:38.130352 240415 kv/txn.go:596  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200319 18:12:38.131750 240415 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200319 18:12:38.139513 240415 kv/txn.go:596  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200319 18:12:38.144704 240415 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200319 18:12:38.186000 240634 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W200319 18:12:38.193465 240293 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200319 18:12:38.249296 240693 util/stop/stopper.go:539  quiescing
W200319 18:12:38.255752 240558 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
W200319 18:12:38.257451 240742 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
W200319 18:12:38.258198 240676 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200319 18:12:38.272273 240693 util/stop/stopper.go:539  quiescing
I200319 18:12:38.281520 240693 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.53s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584641558.184584784,0 pro=1584641557.284689554,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584641558.184584784,0 pro=1584641557.284689554,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584641558.634798070,0 pro=1584641557.734908971,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584641558.184584784,0 pro=1584641557.284689554,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584641558.184584784,0 pro=1584641557.284689554,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584641558.184584784,0 pro=1584641557.284689554,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584641558.184584784,0 pro=1584641557.284689554,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584641558.634798070,0 pro=1584641557.734908971,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1821259&tab=buildLog) on [master@7241972bb2067b2a8153282b478ef09c224b99cb](https://github.com/cockroachdb/cockroach/commits/7241972bb2067b2a8153282b478ef09c224b99cb):

```
=== RUN   TestRangeInfo
I200320 19:15:14.774359 241488 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34417" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200320 19:15:14.922369 241488 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200320 19:15:14.923442 241488 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:45099" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200320 19:15:14.931673 241586 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:34417
I200320 19:15:15.036446 241488 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200320 19:15:15.068060 241488 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=52098593] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200320 19:15:15.120903 241488 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot ca4b3c91 at applied index 18
I200320 19:15:15.133359 241488 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.05s
I200320 19:15:15.161623 241838 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=ca4b3c91 index=18]
I200320 19:15:15.175999 241838 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=8ms ingestion=4@3ms id=ca4b3c91 index=18]
I200320 19:15:15.200912 241488 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200320 19:15:15.225150 241488 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=5dca1123] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200320 19:15:15.419402 241488 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200320 19:15:15.867718 242080 util/stop/stopper.go:539  quiescing
I200320 19:15:15.868298 242079 util/stop/stopper.go:539  quiescing
W200320 19:15:15.877424 241587 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200320 19:15:15.877736 241587 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200320 19:15:15.880197 241587 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200320 19:15:15.880649 241587 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200320 19:15:15.901839 241859 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W200320 19:15:15.919356 241884 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200320 19:15:15.925267 242078 util/stop/stopper.go:539  quiescing
W200320 19:15:15.932240 241880 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200320 19:15:15.933238 241969 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200320 19:15:15.937142 241987 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200320 19:15:15.939171 242078 util/stop/stopper.go:539  quiescing
I200320 19:15:15.940585 242078 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.29s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584731716.107316995,0 pro=1584731715.207400980,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584731716.107316995,0 pro=1584731715.207400980,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584731716.565932104,0 pro=1584731715.666068968,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584731716.107316995,0 pro=1584731715.207400980,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584731716.107316995,0 pro=1584731715.207400980,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584731716.107316995,0 pro=1584731715.207400980,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584731716.107316995,0 pro=1584731715.207400980,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584731716.565932104,0 pro=1584731715.666068968,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1821463&tab=buildLog) on [master@16f708d4db2292b9436aff00a966b6cf05abd818](https://github.com/cockroachdb/cockroach/commits/16f708d4db2292b9436aff00a966b6cf05abd818):

```
=== RUN   TestRangeInfo
I200320 22:36:23.797868 241093 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:32873" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200320 22:36:23.986986 241093 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200320 22:36:23.989210 241093 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:37493" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200320 22:36:23.991713 241333 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:32873
I200320 22:36:24.099045 241093 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200320 22:36:24.126352 241093 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=7e883f79] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200320 22:36:24.180789 241093 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 3d7a2418 at applied index 18
I200320 22:36:24.187328 241093 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200320 22:36:24.197397 241325 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=3d7a2418 index=18]
I200320 22:36:24.207783 241325 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=10ms ingestion=4@4ms id=3d7a2418 index=18]
I200320 22:36:24.228660 241093 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200320 22:36:24.280405 241093 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=0be6d56c] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200320 22:36:24.413324 241093 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200320 22:36:25.050673 241669 util/stop/stopper.go:539  quiescing
I200320 22:36:25.051362 241524 util/stop/stopper.go:539  quiescing
W200320 22:36:25.058947 241107 kv/kvclient/kvcoord/txn_interceptor_heartbeater.go:400  node unavailable; try another peer
W200320 22:36:25.061953 241107 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200320 22:36:25.062440 241107 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200320 22:36:25.075988 241301 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200320 22:36:25.101818 241561 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W200320 22:36:25.112505 241347 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):?: no handler registered for (n1,s1):?
W200320 22:36:25.131051 241557 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
W200320 22:36:25.112597 241561 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W200320 22:36:25.140310 241401 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200320 22:36:25.154256 241523 util/stop/stopper.go:539  quiescing
W200320 22:36:25.168177 241411 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200320 22:36:25.168814 241358 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200320 22:36:25.172389 241523 util/stop/stopper.go:539  quiescing
I200320 22:36:25.174282 241523 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.49s)
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584743785.122263258,0 pro=1584743784.222394004,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584743785.122263258,0 pro=1584743784.222394004,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584743785.122263258,0 pro=1584743784.222394004,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584743785.572586113,0 pro=1584743784.674297046,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1821544&tab=buildLog) on [master@16f708d4db2292b9436aff00a966b6cf05abd818](https://github.com/cockroachdb/cockroach/commits/16f708d4db2292b9436aff00a966b6cf05abd818):

```
=== RUN   TestRangeInfo
I200321 00:50:04.859649 236909 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:45189" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200321 00:50:04.964985 236909 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200321 00:50:04.966491 236909 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:41029" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200321 00:50:04.968152 237073 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:45189
I200321 00:50:05.094370 236909 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200321 00:50:05.123824 236909 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=34129923] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200321 00:50:05.151325 236909 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot f0fa6115 at applied index 18
I200321 00:50:05.155346 236909 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200321 00:50:05.179471 237226 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200321 00:50:05.180766 237242 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=f0fa6115 index=18]
I200321 00:50:05.204773 237242 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=23ms ingestion=4@20ms id=f0fa6115 index=18]
I200321 00:50:05.269604 236909 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200321 00:50:05.307840 236909 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=7c0e1834] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200321 00:50:05.407969 236909 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200321 00:50:06.091965 237561 util/stop/stopper.go:539  quiescing
I200321 00:50:06.093127 237562 util/stop/stopper.go:539  quiescing
I200321 00:50:06.137948 237560 util/stop/stopper.go:539  quiescing
W200321 00:50:06.141802 237357 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200321 00:50:06.142494 237239 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200321 00:50:06.144343 237391 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200321 00:50:06.145004 237305 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200321 00:50:06.148633 237560 util/stop/stopper.go:539  quiescing
I200321 00:50:06.150632 237560 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.39s)
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584751806.187678458,0 pro=1584751805.289918732,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584751806.187678458,0 pro=1584751805.289918732,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584751806.187678458,0 pro=1584751805.289918732,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584751806.646261373,0 pro=1584751805.746371208,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1821781&tab=buildLog) on [master@322f83364bc0807c3147af0631cb7317c584ce43](https://github.com/cockroachdb/cockroach/commits/322f83364bc0807c3147af0631cb7317c584ce43):

```
=== RUN   TestRangeInfo
I200321 05:56:43.307420 236234 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46037" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200321 05:56:43.435296 236234 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200321 05:56:43.436376 236234 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:40313" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200321 05:56:43.444926 236317 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:46037
I200321 05:56:43.523703 236234 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200321 05:56:43.533788 236234 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=8ebfb42f] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200321 05:56:43.552495 236234 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 481a7f60 at applied index 18
I200321 05:56:43.556850 236234 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200321 05:56:43.565443 236553 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=481a7f60 index=18]
I200321 05:56:43.573121 236553 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=7ms ingestion=4@3ms id=481a7f60 index=18]
I200321 05:56:43.590398 236234 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200321 05:56:43.607462 236234 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=69ed9829] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200321 05:56:43.680679 236234 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200321 05:56:44.185779 236776 util/stop/stopper.go:539  quiescing
I200321 05:56:44.186593 236777 util/stop/stopper.go:539  quiescing
I200321 05:56:44.251289 236775 util/stop/stopper.go:539  quiescing
W200321 05:56:44.261981 236519 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200321 05:56:44.268180 236521 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200321 05:56:44.269433 236561 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
I200321 05:56:44.271935 236775 util/stop/stopper.go:539  quiescing
I200321 05:56:44.273777 236775 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.07s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584770204.190166453,0 pro=1584770203.292463127,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584770204.190166453,0 pro=1584770203.292463127,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584770204.640428262,0 pro=1584770203.740526878,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584770204.190166453,0 pro=1584770203.292463127,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584770204.190166453,0 pro=1584770203.292463127,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584770204.190166453,0 pro=1584770203.292463127,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584770204.190166453,0 pro=1584770203.292463127,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584770204.640428262,0 pro=1584770203.740526878,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1822502&tab=buildLog) on [master@322f83364bc0807c3147af0631cb7317c584ce43](https://github.com/cockroachdb/cockroach/commits/322f83364bc0807c3147af0631cb7317c584ce43):

```
=== RUN   TestRangeInfo
I200321 09:54:27.134549 243181 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:35903" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200321 09:54:27.294924 243181 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200321 09:54:27.296453 243181 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:36259" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200321 09:54:27.300028 243308 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:35903
I200321 09:54:27.419705 243181 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200321 09:54:27.441933 243181 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=df4720fc] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200321 09:54:27.492502 243181 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot faa2e10f at applied index 19
I200321 09:54:27.495345 243181 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200321 09:54:27.523807 243357 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=faa2e10f index=19]
I200321 09:54:27.544579 243357 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=20ms ingestion=4@6ms id=faa2e10f index=19]
I200321 09:54:27.606510 243181 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200321 09:54:27.655497 243181 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=8be6e0bf] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200321 09:54:27.803934 243181 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200321 09:54:28.314275 243754 util/stop/stopper.go:539  quiescing
I200321 09:54:28.321650 243755 util/stop/stopper.go:539  quiescing
W200321 09:54:28.325036 243278 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200321 09:54:28.325709 243278 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
I200321 09:54:28.367673 243753 util/stop/stopper.go:539  quiescing
W200321 09:54:28.373819 243670 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200321 09:54:28.374697 243669 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200321 09:54:28.375546 243440 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200321 09:54:28.378816 243753 util/stop/stopper.go:539  quiescing
I200321 09:54:28.380429 243753 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.36s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584784468.471720629,0 pro=1584784467.571877609,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584784468.471720629,0 pro=1584784467.571877609,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584784468.948119802,0 pro=1584784468.048180749,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584784468.471720629,0 pro=1584784467.571877609,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584784468.471720629,0 pro=1584784467.571877609,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584784468.471720629,0 pro=1584784467.571877609,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584784468.471720629,0 pro=1584784467.571877609,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584784468.948119802,0 pro=1584784468.048180749,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1822513&tab=buildLog) on [master@322f83364bc0807c3147af0631cb7317c584ce43](https://github.com/cockroachdb/cockroach/commits/322f83364bc0807c3147af0631cb7317c584ce43):

```
=== RUN   TestRangeInfo
I200321 11:34:17.244291 233264 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38767" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200321 11:34:17.327554 233264 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200321 11:34:17.328870 233264 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:45967" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200321 11:34:17.331360 233542 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:38767
I200321 11:34:17.407238 233264 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200321 11:34:17.417898 233264 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=f4e63e06] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200321 11:34:17.458042 233264 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 9177d23e at applied index 19
I200321 11:34:17.464438 233264 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200321 11:34:17.474629 233547 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=9177d23e index=19]
I200321 11:34:17.483094 233622 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200321 11:34:17.483929 233547 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=9ms ingestion=4@5ms id=9177d23e index=19]
I200321 11:34:17.492313 233264 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200321 11:34:17.509331 233264 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=602c3887] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200321 11:34:17.552625 233264 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200321 11:34:17.840762 233784 util/stop/stopper.go:539  quiescing
I200321 11:34:17.841119 233785 util/stop/stopper.go:539  quiescing
W200321 11:34:17.844758 233527 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200321 11:34:17.845100 233527 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
I200321 11:34:17.849062 233474 kv/kvserver/queue.go:578  [s2,r2/2:{a-/Max}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200321 11:34:17.887166 233783 util/stop/stopper.go:539  quiescing
W200321 11:34:17.891354 233600 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200321 11:34:17.891613 233704 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200321 11:34:17.892526 233412 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200321 11:34:17.893685 233395 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200321 11:34:17.895090 233783 util/stop/stopper.go:539  quiescing
I200321 11:34:17.896592 233783 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (0.76s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584790458.117236866,0 pro=1584790457.217352979,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584790458.117236866,0 pro=1584790457.217352979,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584790458.577411485,0 pro=1584790457.677498841,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584790458.117236866,0 pro=1584790457.217352979,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584790458.117236866,0 pro=1584790457.217352979,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584790458.117236866,0 pro=1584790457.217352979,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584790458.117236866,0 pro=1584790457.217352979,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584790458.577411485,0 pro=1584790457.677498841,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1822524&tab=buildLog) on [master@322f83364bc0807c3147af0631cb7317c584ce43](https://github.com/cockroachdb/cockroach/commits/322f83364bc0807c3147af0631cb7317c584ce43):

```
=== RUN   TestRangeInfo
I200321 13:14:50.236992 236238 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:45527" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200321 13:14:50.335450 236238 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200321 13:14:50.337216 236238 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:39471" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200321 13:14:50.346061 236395 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:45527
I200321 13:14:50.440684 236238 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200321 13:14:50.452607 236238 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=3c5295a6] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200321 13:14:50.472745 236238 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot e70c84b4 at applied index 18
I200321 13:14:50.478396 236238 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I200321 13:14:50.490625 236509 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=e70c84b4 index=18]
I200321 13:14:50.511576 236509 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=20ms ingestion=4@7ms id=e70c84b4 index=18]
I200321 13:14:50.563843 236238 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200321 13:14:50.575255 236238 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=5d429a5f] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200321 13:14:50.629955 236238 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200321 13:14:51.211497 236752 util/stop/stopper.go:539  quiescing
I200321 13:14:51.213843 236751 util/stop/stopper.go:539  quiescing
W200321 13:14:51.218468 236469 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200321 13:14:51.279569 236750 util/stop/stopper.go:539  quiescing
W200321 13:14:51.287681 236498 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200321 13:14:51.288955 236581 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200321 13:14:51.289617 236594 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200321 13:14:51.290670 236667 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200321 13:14:51.294654 236750 util/stop/stopper.go:539  quiescing
I200321 13:14:51.296667 236750 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.16s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584796491.111409488,0 pro=1584796490.211532976,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584796491.111409488,0 pro=1584796490.211532976,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584796491.572823521,0 pro=1584796490.676186358,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584796491.111409488,0 pro=1584796490.211532976,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584796491.111409488,0 pro=1584796490.211532976,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584796491.111409488,0 pro=1584796490.211532976,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584796491.111409488,0 pro=1584796490.211532976,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584796491.572823521,0 pro=1584796490.676186358,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1823047&tab=buildLog) on [master@504b9998c51866950785db5c92ac9cfa9401a062](https://github.com/cockroachdb/cockroach/commits/504b9998c51866950785db5c92ac9cfa9401a062):

```
=== RUN   TestRangeInfo
I200323 17:34:46.922006 237539 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34387" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200323 17:34:46.994104 237539 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200323 17:34:46.998266 237539 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:42769" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200323 17:34:47.004032 237978 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:34387
I200323 17:34:47.081453 237539 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200323 17:34:47.093400 237539 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=709ad343] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200323 17:34:47.116470 237539 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot cb6be860 at applied index 18
I200323 17:34:47.121035 237539 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200323 17:34:47.131459 237858 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=cb6be860 index=18]
I200323 17:34:47.140231 237861 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200323 17:34:47.146320 237858 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=14ms ingestion=4@11ms id=cb6be860 index=18]
I200323 17:34:47.163353 237539 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200323 17:34:47.212217 237539 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=f72cdc45] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200323 17:34:47.306847 237539 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200323 17:34:47.840300 238264 util/stop/stopper.go:539  quiescing
W200323 17:34:47.844463 237797 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200323 17:34:47.845987 237797 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
I200323 17:34:47.846355 238265 util/stop/stopper.go:539  quiescing
I200323 17:34:47.848631 237931 kv/kvserver/queue.go:578  [s2,r2/2:{a-/Max}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200323 17:34:47.906154 237949 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200323 17:34:47.941583 238263 util/stop/stopper.go:539  quiescing
W200323 17:34:47.954688 238098 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200323 17:34:47.955517 238095 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200323 17:34:47.959070 237907 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
W200323 17:34:47.959815 237974 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
I200323 17:34:47.961739 238263 util/stop/stopper.go:539  quiescing
I200323 17:34:47.963916 238263 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.13s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584984887.799864765,0 pro=1584984886.899973076,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584984887.799864765,0 pro=1584984886.899973076,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584984888.252353705,0 pro=1584984887.352466860,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584984887.799864765,0 pro=1584984886.899973076,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584984887.799864765,0 pro=1584984886.899973076,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584984887.799864765,0 pro=1584984886.899973076,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584984887.799864765,0 pro=1584984886.899973076,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584984888.252353705,0 pro=1584984887.352466860,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1823184&tab=buildLog) on [master@885821a6126cd71d69355d9f0360d4e5a68f6060](https://github.com/cockroachdb/cockroach/commits/885821a6126cd71d69355d9f0360d4e5a68f6060):

```
=== RUN   TestRangeInfo
I200323 18:39:16.848611 238411 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:33919" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200323 18:39:16.934664 238411 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200323 18:39:16.938590 238411 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:37361" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200323 18:39:16.940066 238541 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:33919
I200323 18:39:16.987133 238411 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200323 18:39:16.997234 238411 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=9bfee1a9] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200323 18:39:17.022433 238411 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 42ad8b50 at applied index 18
I200323 18:39:17.025300 238411 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200323 18:39:17.042085 238693 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=42ad8b50 index=18]
I200323 18:39:17.048650 238693 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=6ms ingestion=4@2ms id=42ad8b50 index=18]
I200323 18:39:17.071826 238411 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200323 18:39:17.083960 238411 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=e1403329] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200323 18:39:17.125492 238411 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200323 18:39:17.501524 238871 util/stop/stopper.go:539  quiescing
I200323 18:39:17.502294 238872 util/stop/stopper.go:539  quiescing
W200323 18:39:17.504791 238505 kv/kvserver/store.go:1541  [s1,r1/1:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200323 18:39:17.507861 238620 kv/kvserver/queue.go:578  [s2,r1/2:{/Min-a}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200323 18:39:17.518117 238646 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200323 18:39:17.518864 238646 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200323 18:39:17.529106 238646 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200323 18:39:17.530565 238646 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
I200323 18:39:17.603615 238870 util/stop/stopper.go:539  quiescing
W200323 18:39:17.608690 238720 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
W200323 18:39:17.612375 238661 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200323 18:39:17.612889 238723 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200323 18:39:17.614966 238870 util/stop/stopper.go:539  quiescing
I200323 18:39:17.618465 238870 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (0.89s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584988757.722894443,0 pro=1584988756.823009862,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584988757.722894443,0 pro=1584988756.823009862,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584988758.175541519,0 pro=1584988757.275724754,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584988757.722894443,0 pro=1584988756.823009862,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584988757.722894443,0 pro=1584988756.823009862,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584988757.722894443,0 pro=1584988756.823009862,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584988757.722894443,0 pro=1584988756.823009862,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584988758.175541519,0 pro=1584988757.275724754,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1823272&tab=buildLog) on [master@00605e53c92766fdf961dba3c6f8d1970e6d6a8f](https://github.com/cockroachdb/cockroach/commits/00605e53c92766fdf961dba3c6f8d1970e6d6a8f):

```
=== RUN   TestRangeInfo
I200323 19:59:11.546536 237774 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46819" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200323 19:59:11.666263 237774 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200323 19:59:11.667935 237774 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:44479" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200323 19:59:11.676273 238125 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:46819
I200323 19:59:11.766470 237774 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200323 19:59:11.807433 237774 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=e6fed3ae] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200323 19:59:11.917841 237774 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot b19a0bdd at applied index 18
I200323 19:59:11.929244 237774 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.07s
I200323 19:59:11.942668 238033 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=b19a0bdd index=18]
I200323 19:59:11.964574 238033 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=12ms ingestion=4@8ms id=b19a0bdd index=18]
I200323 19:59:12.041228 237774 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200323 19:59:12.077843 237774 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=825f3afc] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200323 19:59:12.244789 237774 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200323 19:59:12.867631 238371 util/stop/stopper.go:539  quiescing
I200323 19:59:12.869810 238370 util/stop/stopper.go:539  quiescing
I200323 19:59:12.918151 238369 util/stop/stopper.go:539  quiescing
W200323 19:59:12.929752 238243 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200323 19:59:12.931934 237890 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
W200323 19:59:12.932183 238325 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200323 19:59:12.943949 238369 util/stop/stopper.go:539  quiescing
I200323 19:59:12.949487 238369 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.55s)
    client_replica_test.go:1573: on scan reply, expected [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584993552.872556430,0 pro=1584993551.972697413,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584993552.872556430,0 pro=1584993551.972697413,0}]; got [{Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584993553.341717129,0 pro=1584993552.441826449,0} {Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584993552.872556430,0 pro=1584993551.972697413,0}]
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584993552.872556430,0 pro=1584993551.972697413,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584993552.872556430,0 pro=1584993551.972697413,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584993552.872556430,0 pro=1584993551.972697413,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584993553.341717129,0 pro=1584993552.441826449,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1823358&tab=buildLog) on [master@dd8b27c3d757880ff41517f13487a22f300740f5](https://github.com/cockroachdb/cockroach/commits/dd8b27c3d757880ff41517f13487a22f300740f5):

```
=== RUN   TestRangeInfo
I200323 21:09:56.427445 238421 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:39331" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200323 21:09:56.517388 238421 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200323 21:09:56.518463 238421 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:39527" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200323 21:09:56.527532 238403 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:39331
I200323 21:09:56.641654 238421 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200323 21:09:56.654323 238421 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=b1f9fc5c] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200323 21:09:56.696490 238421 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 3198b504 at applied index 18
I200323 21:09:56.706194 238421 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200323 21:09:56.776503 238687 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=3198b504 index=18]
I200323 21:09:56.803021 238706 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200323 21:09:56.837957 238687 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=55ms ingestion=4@24ms id=3198b504 index=18]
I200323 21:09:56.881441 238421 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200323 21:09:56.955668 238421 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=f8943839] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200323 21:09:57.084383 238421 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200323 21:09:57.482351 238938 util/stop/stopper.go:539  quiescing
I200323 21:09:57.484041 238939 util/stop/stopper.go:539  quiescing
I200323 21:09:57.499407 238612 kv/kvserver/queue.go:578  [s2,r2/2:{a-/Max}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200323 21:09:57.518673 238548 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200323 21:09:57.519780 238548 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
I200323 21:09:57.548496 238937 util/stop/stopper.go:539  quiescing
W200323 21:09:57.551918 238797 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200323 21:09:57.553627 238685 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200323 21:09:57.554073 238705 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200323 21:09:57.554155 238857 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200323 21:09:57.561730 238937 util/stop/stopper.go:539  quiescing
I200323 21:09:57.563438 238937 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.24s)
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584997797.750732254,0 pro=1584997796.850866651,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584997797.750732254,0 pro=1584997796.850866651,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584997797.750732254,0 pro=1584997796.850866651,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1584997798.201617597,0 pro=1584997797.301702930,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRangeInfo failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1823593&tab=buildLog) on [master@e280cdb84567eca122bb7538f28d3b0c1ec72551](https://github.com/cockroachdb/cockroach/commits/e280cdb84567eca122bb7538f28d3b0c1ec72551):

```
=== RUN   TestRangeInfo
I200323 22:43:42.759944 240306 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42545" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200323 22:43:42.868243 240306 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200323 22:43:42.871344 240306 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:38275" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200323 22:43:42.881792 240636 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:42545
I200323 22:43:43.006381 240306 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200323 22:43:43.044802 240306 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=cd619f87] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200323 22:43:43.093985 240306 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot d388c5f5 at applied index 19
I200323 22:43:43.104559 240306 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 78, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200323 22:43:43.147217 240680 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=d388c5f5 index=19]
I200323 22:43:43.149910 240758 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200323 22:43:43.157870 240680 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=10ms ingestion=4@7ms id=d388c5f5 index=19]
I200323 22:43:43.172591 240306 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200323 22:43:43.234727 240306 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=95faf5cb] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200323 22:43:43.346223 240306 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200323 22:43:43.904005 240942 util/stop/stopper.go:539  quiescing
I200323 22:43:43.904952 240943 util/stop/stopper.go:539  quiescing
I200323 22:43:43.911645 240940 kv/kvserver/queue.go:1189  [replicate] purgatory is now empty
W200323 22:43:43.951138 240664 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200323 22:43:43.951463 240664 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200323 22:43:43.955970 240664 kv/txn.go:603  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200323 22:43:43.957213 240664 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200323 22:43:43.973897 240623 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200323 22:43:44.031655 240941 util/stop/stopper.go:539  quiescing
W200323 22:43:44.036260 240740 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
W200323 22:43:44.043585 240699 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200323 22:43:44.045308 240801 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200323 22:43:44.045428 240712 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200323 22:43:44.053257 240941 util/stop/stopper.go:539  quiescing
I200323 22:43:44.055303 240941 util/stop/stopper.go:539  quiescing
--- FAIL: TestRangeInfo (1.43s)
    client_replica_test.go:1598: on reverse scan reply, expected [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1585003424.084498850,0 pro=1585003423.184610131,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1585003424.084498850,0 pro=1585003423.184610131,0}]; got [{Desc:r2:{a-/Max} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1585003424.084498850,0 pro=1585003423.184610131,0} {Desc:r1:{/Min-a} [(n1,s1):1, (n2,s2):2, next=3, gen=3] Lease:repl=(n1,s1):1 seq=1 start=0,0 exp=1585003424.534651937,0 pro=1585003423.634748362,0}]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRangeInfo PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRangeInfo.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -1454,6 +1454,7 @@ func TestRangeInfo(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	storeCfg := kvserver.TestStoreConfig(nil /* clock */)
 	storeCfg.TestingKnobs.DisableMergeQueue = true
+	storeCfg.Clock = nil // manual clock
 	mtc := &multiTestContext{
 		storeConfig: &storeCfg,
 		// This test was written before the multiTestContext started creating many

